### PR TITLE
Allow setting IBuildStrategy on BuildConfig

### PR DESCRIPTION
**Pull request by aslakknutsen**
Not sure if this was the idea of the // FIXME, but it in general works.

Any bigger plans beyond this? If not I'll implement similar for BuildSource and BuildTriggers

### CODE REVIEW

File: src/main/java/com/openshift/internal/restclient/model/BuildConfig.java [hunk 1]
```diff
@@ -142,6 +200,14 @@ public void setOutput(DockerImageURI imageUri){
 		return null;
 	}
 
+	private void storeEnvironmentVars(final String [] key, ModelNode root, Map<String, String> envs){
```
Comment (jcantrill) [inline src/main/java/com/openshift/internal/restclient/model/BuildConfig.java]: @aslakknutsen consider making this variable accept a key that will be used to look up a path similiar to the existing convenience "get" methods

Comment (aslakknutsen) [inline src/main/java/com/openshift/internal/restclient/model/BuildConfig.java]: Right, was wondering about that. I ended up just implementing it as how 'loadEnvrionmentVars' was implemented :)

Comment (aslakknutsen) [inline src/main/java/com/openshift/internal/restclient/model/BuildConfig.java]: @jcantrill Should be handled in this commit: https://github.com/aslakknutsen/openshift-restclient-java/commit/9b738fb5a8751774f6c450a6026a68ceba057974 

Named them setEnvMap and getEnvMap to differentiate between Kubernetes EnvVar(using name and value) type used here and just  JSON Maps used in e.g. Labels (asMap())

**Comment by jcantrill** [comment]
@aslakknutsen LGTM.  Merged

**Pull request merged**